tests/models/mega/test_modeling_mega.py

```diff
@@ -0,0 +1,592 @@
+
+        input_mask = None
+
+        self.model_tester.create_and_check_model_as_decoder(
+            config,
+            input_ids,
+            token_type_ids,
+            input_mask,
+            sequence_labels,
+            token_labels,
+            choice_labels,
+            encoder_hidden_states,
+            encoder_attention_mask,
+        )
+
+    def test_for_causal_lm(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs_for_decoder()
+        self.model_tester.create_and_check_for_causal_lm(*config_and_inputs)
+
+    def test_decoder_model_past_with_large_inputs(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs_for_decoder()
+        self.model_tester.create_and_check_decoder_model_past_large_inputs(*config_and_inputs)
+
+    def test_for_masked_lm(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_masked_lm(*config_and_inputs)
+
+    def test_for_token_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_token_classification(*config_and_inputs)
+
+    def test_for_multiple_choice(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_multiple_choice(*config_and_inputs)
+
+    def test_for_question_answering(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_question_answering(*config_and_inputs)
+
+    def test_for_bidirectionality(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_bidirectionality(*config_and_inputs)
+
+    def test_for_chunking_shorter_sequence(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.check_chunking_shorter_sequence(*config_and_inputs)
+
+    def test_for_chunking_longer_sequence(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.check_chunking_longer_sequence(*config_and_inputs)
+
+    def test_for_element_attention(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.check_element_attention(*config_and_inputs)
+
+    def test_for_sequence_length_beyond_max_positions(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.check_sequence_length_beyond_max_positions(*config_and_inputs)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in MEGA_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = MegaModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+@require_torch
+class MegaModelIntegrationTest(TestCasePlus):
```

Comment:
Okay! Are there some examples of the tests I should add? I followed [RoBERTa's integration tests](https://github.com/huggingface/transformers/blob/main/tests/models/roberta/test_modeling_roberta.py#L527) and removed the sequence classification test because there is currently no pretrained sequence classifier for Mega

Reply:
Also on this: there are some generation tests in the common modeling tests, so I assumed that since RoBERTa doesn't add any further generation tests, this would be okay for Mega

Reply:
Roberta is an old model, we try to have as many test as possible especially for new models! Take inspiration from T5 or OPt!

Reply:
Okay I'll take a look and add what I can!

Reply:
It looks like OPT is the closer of the two models, since it can behave as a standalone encoder or decoder like MEGA/BERT/etc.

I added the `test_generate_fp16` test from OPT, and it's passing locally on CPU. The other generation tests for OPT are testing the correctness of generated values from existing pretrained LMs (for example, [this one](https://github.com/huggingface/transformers/blob/main/tests/models/opt/test_modeling_opt.py#L435)). I held off on porting these over because there is no pretrained causal LM for Mega as of now.

Reply:
Thanks for adding that, looks good to me